src/AWS.Deploy.Common/Recipes/Validation/OptionSettingItemValidators/FileValidator.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.IO;
+using AWS.Deploy.Common.IO;
+
+namespace AWS.Deploy.Common.Recipes.Validation
+{
+    /// <summary>
+    /// Validates that a recipe or deployment bundle option with a FilePath typehint points to an actual file.
+    /// This can either be an absolute path to the file or relative to the project path for the current recommendation.
+    /// </summary>
+    public class FileValidator : IOptionSettingItemValidator
+    {
+        public static readonly string defaultValidationFailedMessage = "The specified file does not exist";
+
+        public string ValidationFailedMessage { get; set; } = defaultValidationFailedMessage;
+
+        ValidationResult IOptionSettingItemValidator.Validate(object input, Recommendation recommendation)
+        {
+            IFileManager fileManager = new FileManager();
```

Comment:
can you pass in IFileManager using dependency injection instead of instantiating it here?

Reply:
Based on this it doesn't seem like we're set up to inject into the `IOptionSettingItemValidator`:
https://github.com/aws/aws-dotnet-deploy/blob/f2e5db538a032be70de1dc3953b297528a62afeb/docs/OptionSettingsItems-input-validation.md?plain=1#L142-L146

Reply:
Switched to injection now that it's possible after https://github.com/aws/aws-dotnet-deploy/pull/560 has been merged.